### PR TITLE
Try to fix #26995

### LINKED ISSUE #26995: Issue with artifactory.downloaded and snapshot artifacts
Trying to utilize `artifactory.downloaded` to deploy an assembly artifact to a minion. I've placed a copy of an example along with steps I've taken to try and diagnose at https://gist.github.com/jbouse/bd5260e2a7635867f973 including `salt --versions-report` output. This is slightly different from #19879 in that I'm not getting the error that artifactory.downloaded is unavailable. Instead I get the following:

```
$ salt minion-id state.sls test
minion-id:
----------
           ID: myco-artifact
     Function: artifactory.downloaded
       Result: False
      Comment: An exception occurred in this state: Traceback (most recent call last):
            File "/usr/lib/python2.6/site-packages/salt/state.py", line 1563, in call
              self.verify_ret(ret)
            File "/usr/lib/python2.6/site-packages/salt/state.py", line 796, in verify_ret
              'Malformed state return, return must be a dict'
          SaltException: Malformed state return, return must be a dict
      Started: 12:44:50.671013
     Duration: 62.355 ms
```

If I use salt-call and call artifactory.get_snapshot with both version and snapshot_version then I am successful but calling artifactory.downloaded in the state file which is the same as calling with snapshot_version=None fails and there doesn't appear to be anyway to call artifactory.downloaded and pass snapshot_version along as the internal call to artifactory.get_snapshot doesn't pass a snapshot_version parameter.

Calling artifactory.get_snapshot in attempt to duplicate the call from within artifactory.downloaded I get the following traceback:

```
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    salt_call()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 227, in salt_call
    client.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/call.py", line 69, in run
    caller.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/caller.py", line 236, in run
    ret = self.call()
  File "/usr/lib/python2.6/site-packages/salt/cli/caller.py", line 138, in call
    ret['return'] = func(*args, **kwargs)
  File "/usr/lib/python2.6/site-packages/salt/modules/artifactory.py", line 73, in get_snapshot
    snapshot_url, file_name = _get_snapshot_url(artifactory_url, repository, group_id, artifact_id, version, packaging, snapshot_version)
  File "/usr/lib/python2.6/site-packages/salt/modules/artifactory.py", line 117, in _get_snapshot_url
    snapshot_version_metadata = _get_snapshot_version_metadata(artifactory_url=artifactory_url, repository=repository, group_id=group_id, artifact_id=artifact_id, version=version)
  File "/usr/lib/python2.6/site-packages/salt/modules/artifactory.py", line 239, in _get_snapshot_version_metadata
    assert version == metadata.find('version').text
AssertionError
```

### PR DESCRIPTION
**Pull request by SuperPommeDeTerre**
### What does this PR do?
Attempt to fix #26995.
When downloading a SNAPSHOT artefact from Artifactory, the `maven-metadata.xml` file could not be present if the artifact is unique. So, the snapshot_version should be equal to `version`.

### What issues does this PR fix or reference?
#26995

### Previous Behavior
State `artifactory.downloaded` failed when `maven-metadata.xml` couln't be retrived from Artifactory for a SNAPSHOT artifact.

### New Behavior
The artifactory module assume `snapshot_version=version` when the metadata couldn't be retrieved.

### Tests written?
No

### CODE REVIEW

File: salt/modules/artifactory.py [hunk 1]
```diff
@@ -202,45 +202,48 @@ def _get_snapshot_url(artifactory_url, repository, group_id, artifact_id, versio
...
+                            group_id=group_id,
+                            artifact_id=artifact_id,
+                            packaging=packaging,
+                            classifier=classifier,
+                            version=version)
+                raise ArtifactoryError(error_message)
+
+            snapshot_version = snapshot_version_metadata['snapshot_versions'][packaging]
+        except CommandExecutionError as err:
+            log.error('Could not fetch maven-metadat.xml. Assuming snapshot_version=%s.', version)
```
Comment (RostyslavFridman) [inline salt/modules/artifactory.py]: Please fix the error message. `maven-metadat.xml` ->  `maven-metadata.xml`

**Review by RostyslavFridman — CHANGES_REQUESTED** [review]

File: salt/modules/artifactory.py (hunk 1 above)
Comment (SuperPommeDeTerre) [inline salt/modules/artifactory.py]: Fixed in commit 0b666e100bda9b71e39a193e609f302f43918d75.

**Review by RostyslavFridman — APPROVED** [review]

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by cachedout** [comment]
@SuperPommeDeTerre We can't seem to get a clean test run here. Could I trouble you to please rebase this against the head of the 2017.7 branch?

**Comment by SuperPommeDeTerre** [comment]
@cachedout Sure, I'll make this !

**Comment by SuperPommeDeTerre** [comment]
@cachedout Still failing on `integration.modules.test_gem.GemModuleTest.test_update`.
Are there any update to this test ?

**Comment by SuperPommeDeTerre** [comment]
@cachedout Finally found my mistake. It's much better, now !

**Comment by cachedout** [comment]
Thanks very much @SuperPommeDeTerre !

**Pull request merged**